src/two_stage_method.jl

```diff
@@ -98,20 +98,20 @@ function two_stage_method(prob::DiffEqBase.DEProblem,tpoints,data;kernel= :Epane
     kernel_function = decide_kernel(kernel)
     e1 = [1;0]
     e2 = [0;1;0]
-    global du = similar(prob.u0) # have to adjust type for autodifferentiation
-    global sol = Vector{typeof(du)}(undef,n)
+    du = similar(prob.u0) # have to adjust type for autodifferentiation
+    sol = Vector{typeof(du)}(undef,n)
     if mpg_autodiff
-      global du_cache = DiffCache(du)
-      global sol_cache = DiffCache(sol)
+      du_cache = DiffCache(du)
+      sol_cache = DiffCache(sol)
```

Comment:
@ChrisRackauckas this needs to be changed as `sol` is not compatible with the `DiffCache` definition.

Reply:
yes

Reply:
Can you suggest how to resolve this? I have tried a lot of things but haven't been able to get this working.

Reply:
sol should be an array of DiffCache